debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -318,8 +318,15 @@ private void createDataEventsForTable(ChangeEventSourceContext sourceContext, Re
         try (Statement statement = readTableStatement();
                 ResultSet rs = statement.executeQuery(selectStatement.get())) {
 
-            Column[] columns = getColumnsForResultSet(table, rs);
-            final int numColumns = table.columns().size();
+            ResultSetMetaData metaData = rs.getMetaData();
+            Column[] columns = new Column[metaData.getColumnCount()];
+            int greatestColumnPosition = 0;
+            for (int i = 0; i < columns.length; i++) {
```

Comment:
This is very similar to the code in https://github.com/debezium/debezium/pull/1822/files#diff-d9a5092f0eb1f68e8881b17b09f25827R106

Can't it be extracted into a utility method shared?